### PR TITLE
Migrate to WHATWG URL

### PR DESCRIPTION
**Pull request by arniu**

### CODE REVIEW

File: lib/parseURL.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * @param {Object} options
+ * @param {(Object|String)} options.url
+ * @param {Object} [options.query]
+ * @return {URL}
+ */
+module.exports = function parseURL (options) {
+  let parsedURL = null
+  let mergedQuery = {}
+
+  if (typeof options.url === 'object') {
```
Comment (mcollina) [inline lib/parseURL.js]: Is this case possible? From the README, we are not supporting this.

Comment (arniu) [reply lib/parseURL.js]: According the schema, `url` may be `String` or `Object`.

File: lib/request.js [hunk 2]
```diff
@@ -1,29 +1,20 @@
...
+const parseURL = require('./parseURL')
+
+/**
+ * Get hostname:port
+ *
+ * @param {URL} parsedURL
+ * @return {String}
+ */
+function hostHeaderFromURL (parsedURL) {
+  return parsedURL.port
```
Comment (delvedor) [inline lib/request.js]: What are you checking here?
The existence of the `port` value?
If so, I would add a stricter check.

Comment (arniu) [reply lib/request.js]: The new URL's `port` will be empty when using default ports:

```javascript
new URL('http://localhost').port === ''
new URL('http://localhost:80').port === ''
new URL('https://localhost').port === ''
new URL('https://localhost:443').port === ''
```

**Pull request merged**